test/unit/next-babel-loader.test.js

```diff
@@ -192,5 +195,102 @@ describe('next-babel-loader', () => {
+          // complex
+          `import baz2,{yeet}from"c";` +
+          `import baz3,{cats}from"d";` +
+          `import{c,d}from"e";` +
+          `import{e as ee,f as ff}from"f";`
+      )
+      expect(code).toMatchInlineSnapshot(
+        `"import\\"core-js\\";import{foo,bar}from\\"a\\";import baz from\\"b\\";import baz2,{yeet}from\\"c\\";import baz3,{cats}from\\"d\\";import{c,d}from\\"e\\";import{e as ee,f as ff}from\\"f\\";"`
+      )
+    })
+
+    const pageFile = path.resolve(dir, 'pages', 'index.js')
+
+    it('should not drop unused exports by default in a page', async () => {
+      const code = await babel(
+        // effectful
+        `import"core-js";` +
+          // basic
+          `import{foo,bar}from"a";import baz from"b";` +
+          // complex
+          `import baz2,{yeet}from"c";` +
+          `import baz3,{cats}from"d";` +
+          `import{c,d}from"e";` +
+          `import{e as ee,f as ff}from"f";`,
+        { resourcePath: pageFile }
+      )
+      expect(code).toMatchInlineSnapshot(
+        `"import\\"core-js\\";import{foo,bar}from\\"a\\";import baz from\\"b\\";import baz2,{yeet}from\\"c\\";import baz3,{cats}from\\"d\\";import{c,d}from\\"e\\";import{e as ee,f as ff}from\\"f\\";"`
+      )
+    })
+
+    it('should drop unused exports in a modern-apis page', async () => {
+      const code = await babel(
+        // effectful
+        `import"core-js";` +
+          // basic
+          `import{foo,bar}from"a";import baz from"b";` +
+          // complex
+          `import baz2,{yeet}from"c";` +
+          `import baz3,{cats}from"d";` +
+          `import{c,d}from"e";` +
+          `import{e as ee,f as ff}from"f";` +
+          `` +
+          `export function unstable_getStaticProps() { return { props: {} } }`,
+        { resourcePath: pageFile }
+      )
+      expect(code).toMatchInlineSnapshot(`"import\\"core-js\\";"`)
+    })
+
+    it('should keep used exports in a modern-apis page (server)', async () => {
+      const code = await babel(
+        // effectful
+        `import"core-js";` +
+          // basic
+          `import{foo,bar}from"a";import baz from"b";import ooo from"ooo";` +
+          // complex
+          `import baz2,{yeet}from"c";` +
+          `import baz3,{cats}from"d";` +
+          `import{c,d}from"e";` +
+          `import{e as ee,f as ff}from"f";` +
+          `` +
+          `export function unstable_getStaticProps() {foo();baz2();ff();ooo(); return { props: {} }}` +
+          `export default function () { return bar(); }`,
+        { resourcePath: pageFile, isServer: true }
+      )
+      expect(code).toMatchInlineSnapshot(
+        `"import\\"core-js\\";import{foo,bar}from\\"a\\";import ooo from\\"ooo\\";import baz2 from\\"c\\";import{f as ff}from\\"f\\";export function unstable_getStaticProps(){foo();baz2();ff();ooo();return{props:{}};}export default function(){return bar();}"`
+      )
+    })
+
+    it('should keep used exports in a modern-apis page (client)', async () => {
```

Comment:
Do we need to worry about behavior when `require` is used?

Reply:
`require` isn't statically analyzable like a import is, so we can't really handle it. This is meant to be an optimization anyway.

Also, since requires can be inline they'll already be dead code eliminated.

Reply:
Maybe we could ban using `require` if they use the new life cycles, though I'm not sure if we can tell webpack to ignore it.